environment/src/main/java/com/sequenceiq/environment/environment/encryption/EnvironmentEncryptionService.java

```diff
@@ -120,10 +121,9 @@ private CloudContext getCloudContext(EnvironmentDto environment) {
                 .build();
     }
 
-    private List<CloudResource> getResourcesForDeletion(EnvironmentDto environment) {
-        Optional<CloudResource> desCloudResourceOptional = resourceRetriever.findByResourceReferenceAndStatusAndType(
-                environment.getParameters().getAzureParametersDto().getAzureResourceEncryptionParametersDto().getDiskEncryptionSetId(),
-                CommonStatus.CREATED, ResourceType.AZURE_DISK_ENCRYPTION_SET);
+    private List<CloudResource> getResourcesForDeletion(EnvironmentDto environment, String diskEncryptionSetName) {
+        Optional<CloudResource> desCloudResourceOptional = resourceRetriever.findByEnvironmentIdAndNameAndType(
+                environment.getId(), diskEncryptionSetName, ResourceType.AZURE_DISK_ENCRYPTION_SET);
```

Comment:
A different lookup strategy is preferable. Surely it is the most convenient to find the resource by its name, but the naming strategy is an implementation detail in `cloud-azure` that we do not want to depend on.

Therefore, I suggest the following:
* Remove the `diskEncryptionSetName` argument.
* Add a new method to `CloudResourceRetrieverService`: `findByEnvironmentIdAndType(Long environmentId, ResourceType type)`. You shall, in turn, also add `com.sequenceiq.environment.resourcepersister.ResourceService.findByEnvironmentIdAndType()` and `com.sequenceiq.environment.resourcepersister.ResourceRepository.findByEnvironmentIdAndType()` (with the  appropriate `@Query`).
* Look up `desCloudResourceOptional` using `resourceRetriever.findByEnvironmentIdAndType()`. The DES will be either associated with an environment (the global one created here), or potentially later with a cluster (FreeIPA / DL / DH) if we decide to support that. In either case, there will be at most one DES for the respective parent resourceCrn.